### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by kaanozcan** [webpack/enhanced-resolve#152]
Hello I am having a bit of problem not sure if this is a bug and not sure how to tackle it down and provide better context. Any help appreciated.

using: webpack:4.10.2

error:
```
/Users/kaan.ozcan/repos/node-index-page/node_modules/tapable/lib/HookCodeFactory.js:24
				return new Function(this.args({
				       ^

RangeError: Maximum call stack size exceeded
    at new Function (<anonymous>)
    at AsyncSeriesBailHookCodeFactory.create (/Users/kaan.ozcan/repos/node-index-page/node_modules/tapable/lib/HookCodeFactory.js:24:12)
    at AsyncSeriesBailHook.compile (/Users/kaan.ozcan/repos/node-index-page/node_modules/tapable/lib/AsyncSeriesBailHook.js:31:18)
    at AsyncSeriesBailHook._createCall (/Users/kaan.ozcan/repos/node-index-page/node_modules/tapable/lib/Hook.js:24:15)
    at AsyncSeriesBailHook.lazyCompileHook (/Users/kaan.ozcan/repos/node-index-page/node_modules/tapable/lib/Hook.js:34:22)
    at Resolver.doResolve (/Users/kaan.ozcan/repos/node-index-page/node_modules/enhanced-resolve/lib/Resolver.js:229:16)
    at resolver.getHook.tapAsync (/Users/kaan.ozcan/repos/node-index-page/node_modules/enhanced-resolve/lib/TryNextPlugin.js:17:13)
    at AsyncSeriesBailHook.eval [as callAsync] (eval at create (/Users/kaan.ozcan/repos/node-index-page/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:7:1)
    at AsyncSeriesBailHook.lazyCompileHook [as _callAsync] (/Users/kaan.ozcan/repos/node-index-page/node_modules/tapable/lib/Hook.js:35:21)
    at Resolver.doResolve (/Users/kaan.ozcan/repos/node-index-page/node_modules/enhanced-resolve/lib/Resolver.js:229:16)
```

**Comment by kaanozcan** [comment]
So I was using Node api to generate bundle. Now I have converted to cli and my error message changed a bit.

```
/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24
				return new Function(this.args({
				       ^

RangeError: Maximum call stack size exceeded
    at new Function (<anonymous>)
    at AsyncSeriesBailHookCodeFactory.create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12)
    at AsyncSeriesBailHook.compile (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/AsyncSeriesBailHook.js:31:18)
    at AsyncSeriesBailHook._createCall (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/Hook.js:24:15)
    at AsyncSeriesBailHook.lazyCompileHook (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/Hook.js:34:22)
    at Resolver.doResolve (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:229:16)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/TryNextPlugin.js:17:13)
    at AsyncSeriesBailHook.eval [as callAsync] (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:7:1)
    at AsyncSeriesBailHook.lazyCompileHook [as _callAsync] (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/Hook.js:35:21)
    at Resolver.doResolve (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:229:16)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/ModuleKindPlugin.js:19:13)
    at _fn40 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:376:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn39 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:367:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn38 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:358:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn37 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:349:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn36 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:340:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn35 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:331:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn34 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:322:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn33 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:313:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn32 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:304:1)
    at resolver.getHook.tapAsync (/usr/local/lib/node_modules/webpack/node_modules/enhanced-resolve/lib/AliasPlugin.js:53:11)
    at _fn31 (eval at create (/usr/local/lib/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:295:1)
```

Saw the path for the AliasPlugin and commented my aliases(4 in total). Still having the same error.

**Comment by evilebottnawi** [comment]
@kaanozcan Can you create minimum reproducible test repo?

**Comment by kaanozcan** [comment]
@evilebottnawi I'll try to transfer contents of my original repo until I get the same message.

**Comment by sokra** [comment]
maybe fixed by #150

**Comment by kaanozcan** [comment]
@sokra @evilebottnawi 

Hello, It was a issue happened migrating from v3 to v4. I have fixed it trying to prepare repo for test. We had a recursive module resolution in config because we wanted to compile our scoped npm repos on our main project and not their own pipelines to keep the bundle size smaller. We had decided it was a micro optimization and dropped it in favour of maintainability a while ago. I just removed that and everything worked fine. So I'm closing issue.

Since I got your attention may I ask another question please? From v3 to v4 has anything changed about how named chunks works? /* webpackChunkName: "x-page-chunk"*/ comment doesn't seem to work or at least out of box. Thanks.
